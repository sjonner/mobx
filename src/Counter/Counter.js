import React, { Component } from 'react';
import { observer, inject } from 'mobx-react';
import autobind from 'autobind-decorator';

@inject(stores => ({counterStore: stores.store.counter}))
@observer
export default class Counter extends Component {
  render() {
    return (
      <div>
        Count: {this.props.counterStore.count} times!<br />

        <button onClick={this.handleIncrease}>+</button>
        <button onClick={this.handleDecrease}>-</button>
        <button onClick={this.handleReset}>reset</button>
      </div>
    );
  }

  @autobind
  handleIncrease() {
    this.props.counterStore.increaseCounter();
  }

  @autobind
  handleDecrease() {
    this.props.counterStore.decreaseCounter();
  }

  @autobind
  handleReset() {
    this.props.counterStore.resetCounter();
  }
}
