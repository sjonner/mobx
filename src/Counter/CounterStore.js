import { observable, action } from 'mobx';
// import autobind from 'autobind-decorator';

export default class CounterStore {
  @observable count = 0;

  @action
  increaseCounter() {
    this.count++;
  }

  @action
  decreaseCounter() {
    this.count--;
  }

  @action
  resetCounter() {
    this.count = 0;
  }
};
