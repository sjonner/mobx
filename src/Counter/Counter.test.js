import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Counter from './Counter';
import store from '../stores';

beforeEach(() => {
	store.counter.resetCounter();
});

it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(
		<Counter.wrappedComponent counterStore={store.counter} />,
  	div
	);
});

it('renders a snapshot', () => {
  const tree = renderer.create(
  	<Counter.wrappedComponent counterStore={store.counter} />
	).toJSON();

  expect(tree).toMatchSnapshot();
});
