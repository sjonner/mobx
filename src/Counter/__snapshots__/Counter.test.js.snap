// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  Count: 
  0
   times!
  <br />
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    reset
  </button>
</div>
`;
